src/synchronous_product/include/synchronous_product/search_tree.h

```diff
@@ -71,33 +73,55 @@ struct SearchTreeNode
 		assert(parent != nullptr || incoming_actions.empty());
 		assert(!incoming_actions.empty() || parent == nullptr);
 	}
+
+	/** @brief Set the node label.
+	 * @param new_label The new node label
+	 * @param cancel_children If true, cancel children after setting the node label
+	 */
+	void
+	set_label(NodeLabel new_label, bool cancel_children = false)
+	{
+		if (label != NodeLabel::UNLABELED) {
+			return;
+		}
+		label = new_label;
+		if (cancel_children && is_expanded && new_label != NodeLabel::UNLABELED) {
```

Comment:
I was wondering: Is there a use-case or in general, is there a case in which we might pass UNLABELED to this function? If not, I think we can assert that this never happens (I cannot see a reason right now, but perhaps I am missing something).

Reply:
Yes good point, we might want to enforce that we do not label something with `UNLABELED`.

Reply:
Fixed